lib/puma/client.rb

```diff
@@ -110,7 +108,10 @@ def initialize(io, env=nil)
 
     attr_accessor :remote_addr_header, :listener
 
-    def_delegators :@io, :closed?
+    # Remove in Puma 7?
+    def closed?
+      @to_io.closed?
```

Comment:
Why the change from `@io` to `@to_io`?

(Maybe that's better, I don't know)

Reply:
Sorry, I composed a reply, but must have closed the window.

Ruby's TCP & UNIX sockets inherit from `IO`, and `IO#to_io` returns `self`.  Puma's MiniSSL sockets do not inherit from `IO`, just like Ruby's OpenSSL sockets.

Their `closed?` method checks the underlying TCP socket.  So, rather than add another Ruby method call, I used `@to_io` to get the TCP socket.

Note that both `io` and `to_io` are read-only attributes of `Client`, so these can be removed without anyone losing functionality.  As mentioned, that would still be an API change.